### PR TITLE
Feature/final cross fixes

### PR DESCRIPTION
**Pull request by lasote**

### CODE REVIEW

File: conans/client/cmake.py [hunk 1]
```diff
@@ -10,6 +10,23 @@
 from conans import tools
 import os
 import platform
+from conans.util.log import logger
+
+# Deprecated in 0.22
+deprecated_conanfile_param_message = '''
```
Comment (lasote) [inline conans/client/cmake.py]: I don't know why this message is shown in the diff... it was already in develop! right?

File: conans/client/cmake.py [hunk 2]
```diff
@@ -36,19 +53,23 @@ def __init__(self, settings_or_conanfile, generator=None, cmake_system_name=True
...
             self.build = self._build_new
-            self.test = self._test_new
         else:
             raise ConanException("First parameter of CMake() has to be a ConanFile instance.")
 
         self.generator = generator or self._generator()
         self.build_dir = None
         self._cmake_system_name = cmake_system_name
+        # Override by environment
+        if get_env("CONAN_CMAKE_SYSTEM_NAME", "") == "False":
```
Comment (memsharded) [inline conans/client/cmake.py]: 3 get_envs in 4 lines. The logic reads a bit weird.

File: conans/client/cmake.py [hunk 3]
```diff
@@ -117,14 +138,16 @@ def _cmake_cross_build_defines(self, the_os, os_ver):
         ret = OrderedDict()
 
         # SYSTEM NAME
-        env_system_name = get_env("CONAN_CMAKE_SYSTEM_NAME", "")
         os_ver = get_env("CONAN_CMAKE_SYSTEM_VERSION", os_ver)
-        if env_system_name and env_system_name != "False":  # False means not auto-set
-            ret["CMAKE_SYSTEM_NAME"] = env_system_name
+        # String not empty
+        if not isinstance(self._cmake_system_name, bool) and self._cmake_system_name != "":
```
Comment (memsharded) [inline conans/client/cmake.py]: Simplify to ``if self._cmake_system_name``. As it is never True. Maybe the issue is that this flag model is weird

**Comment by memsharded** [comment]
The ``CMAKE_BUILD_SHARED`` thing will be done in a separate PR?

**Comment by lasote** [comment]
Yes, it will be a different pr

File: conans/client/cmake.py (hunk 3 above)
Comment (lasote) [inline conans/client/cmake.py]: Actually the default is True

**Pull request merged**